### ISSUE TITLE
CompactLogin throws an error when auth.url()

### ISSUE THREAD

**Issue by Ivaskuu** [draw-dev/DRAW#46]
```
auth.url(['edit','account','identity','report','privatemessages','read','save','submit','mysubreddits','vote','subscribe'], 'RaNdOm_StRiNg', duration: 'permanent', compactLogin: true).toString();
```
gives me an error:
```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (24594): The following assertion was thrown building LoginPage(dirty, state: _LoginPageState#ff6ad):
I/flutter (24594): The path should end with "authorize?"
I/flutter (24594): 'package:draw/src/auth.dart': Failed assertion: line 377: 'path.endsWith("?")'
I/flutter (24594):
I/flutter (24594): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (24594): more information in this error message to help you determine and fix the underlying cause.
I/flutter (24594): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (24594):   https://github.com/flutter/flutter/issues/new
I/flutter (24594):
I/flutter (24594): When the exception was thrown, this was the stack:
I/flutter (24594): #2      WebAuthenticator.url (package:draw/src/auth.dart:377)
I/flutter (24594): #3      ApiReddit.getAuthUrl (file:///C:/Users/Adrian/Documents/MyApps/focus_reddit/lib/misc/api_reddit.dart:28)
I/flutter (24594): #4      _LoginPageState.loginWebView (file:///C:/Users/Adrian/Documents/MyApps/focus_reddit/lib/pages/login_page.dart:103)
I/flutter (24594): #5      _LoginPageState.build (file:///C:/Users/Adrian/Documents/MyApps/focus_reddit/lib/pages/login_page.dart:51)
I/flutter (24594): #6      StatefulElement.build (package:flutter/src/widgets/framework.dart:3713)
I/flutter (24594): #7      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3625)
I/flutter (24594): #8      Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
I/flutter (24594): #9      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2225)
I/flutter (24594): #10     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame
(package:flutter/src/widgets/binding.dart:621)
I/flutter (24594): #11     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCal
lback (package:flutter/src/rendering/binding.dart:208)
I/flutter (24594): #12     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/bindi
ng.dart:990)
I/flutter (24594): #13     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.da
rt:930)
I/flutter (24594): #14     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.d
art:842)
I/flutter (24594): #15     _invoke (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:120)
I/flutter (24594): #16     _drawFrame (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:109)
I/flutter (24594): (elided 2 frames from class _AssertionError)
I/flutter (24594): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

![image](https://user-images.githubusercontent.com/28832159/38164572-f9a6941e-3506-11e8-8266-c86ad8da10e4.png)

It works fine if I remove `compactLogin: true`

**Comment by bkonyi** [comment]
Hi @Ivaskuu, thanks for filing an issue. I'll look into this today.

**Comment by bkonyi** [comment]
This should be fixed as of #48. I've published the change to pub under version 0.1.4, which should now be available.
